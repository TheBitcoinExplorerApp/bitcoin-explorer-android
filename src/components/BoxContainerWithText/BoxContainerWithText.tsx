// #TODO: WTF I DID ON THIS STYLES
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/require-default-props */
// eslint-disable-next-line import/no-extraneous-dependencies
import { View, Text, StyleSheet } from "react-native";

type BoxContainerWithTextProps = {
  firstText: string;
  secondText: string;
  reverse?: boolean;
  secondTextWhite?: boolean;
  width?: string;
  borderStyles?: {
    borderTopEndRadius?: number;
    borderBottomEndRadius?: number;
    borderTopStartRadius?: number;
    borderBottomStartRadius?: number;
    borderBottomColor?: string;
  };
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#1d2133",
    justifyContent: "space-between",
    paddingVertical: 9,
    paddingHorizontal: 18,
    alignItems: "center",
    height: 48,
  },
  firstText: {
    color: "#FFF",
    fontSize: 15,
  },
});

export default function BoxContainerWithText(props: BoxContainerWithTextProps) {
  const { firstText, secondText, reverse, borderStyles } = props;

  return (
    <View
      style={[
        styles.container,
        {
          flexDirection: reverse ? "row-reverse" : "row",
          width: props.width || "100%",
          borderRadius: 7,
          borderTopEndRadius: borderStyles?.borderTopEndRadius,
          borderBottomEndRadius: borderStyles?.borderBottomEndRadius,
          borderTopStartRadius: borderStyles?.borderTopStartRadius,
          borderBottomStartRadius: borderStyles?.borderBottomStartRadius,
          borderBottomColor: borderStyles?.borderBottomColor,
          borderBottomWidth: borderStyles?.borderBottomColor ? 0.1 : 0,
        },
      ]}
    >
      <Text style={styles.firstText}>{firstText}</Text>
      <Text
        style={{
          color: props.secondTextWhite ? "#FFF" : "#DF7800",
          fontSize: 15,
        }}
      >
        {secondText}
      </Text>
    </View>
  );
}
