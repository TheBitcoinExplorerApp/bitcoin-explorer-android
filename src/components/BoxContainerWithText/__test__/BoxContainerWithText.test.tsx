// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import BoxContainerWithText from "../BoxContainerWithText";

describe("BoxContainerWithText", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<BoxContainerWithText firstText="first" secondText="second" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
