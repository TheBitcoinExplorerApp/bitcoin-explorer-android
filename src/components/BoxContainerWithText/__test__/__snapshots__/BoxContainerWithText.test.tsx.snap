// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxContainerWithText renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#1d2133",
        "height": 48,
        "justifyContent": "space-between",
        "paddingHorizontal": 18,
        "paddingVertical": 9,
      },
      {
        "borderBottomColor": undefined,
        "borderBottomEndRadius": undefined,
        "borderBottomStartRadius": undefined,
        "borderBottomWidth": 0,
        "borderRadius": 7,
        "borderTopEndRadius": undefined,
        "borderTopStartRadius": undefined,
        "flexDirection": "row",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 15,
      }
    }
  >
    first
  </Text>
  <Text
    style={
      {
        "color": "#DF7800",
        "fontSize": 15,
      }
    }
  >
    second
  </Text>
</View>
`;
